plugin/providers/phpipam/data_source_phpipam_subnet.go

```diff
@@ -43,7 +44,19 @@ func dataSourcePHPIPAMSubnetRead(d *schema.ResourceData, meta interface{}) error
 			return err
 		}
 	default:
-		return errors.New("No valid combination of parameters found - need one of subnet_id, subnet_address and subnet_mask, or section_id and (description|description_match|custom_field_filter)")
+    id := d.Id()
+    if len(id) > 0 {
+      subnet_id, err := strconv.Atoi(id)
+      if err != nil {
+        return err
+      }
+		  out[0], err = c.GetSubnetByID(subnet_id)
+		  if err != nil {
+			  return err
+		  }
+    } else {
```

Comment:
I don't get the point of this change. Can you please add a comment why this check is needed?